src/aics-image-viewer/components/App/index.tsx

```diff
@@ -143,435 +134,227 @@ const defaultProps: AppProps = {
-      message = resp.stack;
-    }
-    else if (resp.message) {
-      message = resp.message;
-    }
-    else {
-      message = JSON.stringify(resp);
-    }
-    **/
-    // console.log(message);
-  }
-
-  openImage(imageDirectory: string, doResetViewMode: boolean, keepLuts?: boolean): void {
-    if (imageDirectory === this.state.currentlyLoadedImagePath) {
-      return;
-    }
-    const { baseUrl } = this.props;
-
-    const fullUrl = `${baseUrl}${imageDirectory}`;
-
-    const loadSpec = new LoadSpec();
-    loadSpec.url = fullUrl;
-    loadSpec.subpath = imageDirectory;
-
-    let loader: IVolumeLoader;
-    // if this does NOT end with tif or json,
-    // then we assume it's zarr.
-    if (fullUrl.endsWith(".json")) {
-      loader = new JsonImageInfoLoader();
-    } else if (fullUrl.endsWith(".tif") || fullUrl.endsWith(".tiff")) {
-      loader = new TiffLoader();
-    } else {
-      loader = new OMEZarrLoader();
-    }
-
-    loader
-      .createVolume(loadSpec, (url, v, channelIndex) => {
-        this.onNewChannelData(url, v, channelIndex, keepLuts);
-      })
-      .then((aimg) => {
-        this.onNewVolumeCreated(aimg, imageDirectory, doResetViewMode);
-      })
-      .catch((resp) => this.handleOpenImageException(resp));
-  }
-
-  initializeNewImage(aimg: Volume, newChannelSettings?: ChannelState[]): void {
-    // set alpha slider first time image is loaded to something that makes sense
-    let maskAlpha = this.getInitialAlphaLevel();
-    this.setViewerSettingsInState({ maskAlpha });
-
-    // Here is where we officially hand the image to the volume-viewer
-    this.placeImageInViewer(aimg, newChannelSettings);
-  }
-
-  private getInitialAlphaLevel(): number {
-    const viewerSettingsState = this.state.viewerSettings;
-    const viewerSettingsProps = this.props.viewerSettings;
-    let alphaLevel =
-      viewerSettingsState.imageType === ImageType.segmentedCell && viewerSettingsState.viewMode === ViewMode.threeD
-        ? ALPHA_MASK_SLIDER_3D_DEFAULT
-        : ALPHA_MASK_SLIDER_2D_DEFAULT;
-    // if maskAlpha is defined in viewerConfig then it will override the above
-    if (viewerSettingsProps?.maskAlpha !== undefined) {
-      alphaLevel = viewerSettingsProps.maskAlpha;
-    }
-    return alphaLevel;
-  }
-
-  // set up the Volume into the Viewer using the current initial settings
-  private placeImageInViewer(aimg: Volume, newChannelSettings?: ChannelState[]): void {
-    const { viewerSettings, channelSettings, view3d } = this.state;
-    if (!view3d) {
-      return;
-    }
-    const channelSetting = newChannelSettings || channelSettings;
-    view3d.removeAllVolumes();
-    view3d.addVolume(aimg, {
-      channels: aimg.channel_names.map((name) => {
-        const ch = this.getOneChannelSetting(name, channelSetting);
-        if (!ch) {
-          return {};
+      const activeAxis = activeAxisMap[viewMode];
+
+      // TODO the following behavior/logic is very specific to a particular application's needs
```

Comment:
it looks like this comment goes with the code up above here (line 199-200)

Reply:
oh i see some similar logic below too.  I agree with the comment - it seems pretty weird.

Reply:
I believe you wrote the comment actually